src/libraries/System.Private.CoreLib/src/System/Number.Parsing.cs

```diff
@@ -1939,15 +1937,13 @@ internal static unsafe bool TryParseSingle(ReadOnlySpan<char> value, NumberStyle
         internal static unsafe bool TryStringToNumber(ReadOnlySpan<char> value, NumberStyles styles, ref NumberBuffer number, NumberFormatInfo info)
         {
             Debug.Assert(info != null);
-            fixed (char* stringPointer = &MemoryMarshal.GetReference(value))
+
+            ReadOnlySpan<char> p = value;
+            if (!TryParseNumber(ref p, styles, ref number, info)
+                || ((value.Length - p.Length) < value.Length && !TrailingZeros(value, (value.Length - p.Length))))
```

Comment:
@tannergooding do you know what is this validation? In the original code it is:
|| ((int)(p - stringPointer) < value.Length && !TrailingZeros(value, (int)(p - stringPointer))))

Reply:
It's basically just an `i < value.Length` check. `stringPointer` tracks the beginning of the string and `p` tracks the current index. So `p - stringPointer` gives you the length of the constructed string. `value.Length` tracks the length of the buffer which represents the most digits you can have in the string.

Reply:
Thanks. I think I can reduce it to
`|| (!p.IsEmpty && !TrailingZeros(value, value.Length - p.Length)))`